test/implementations/premint.ts

```diff
@@ -119,6 +119,27 @@ async function shouldControlValidPreMinting(): Promise<void> {
     ).to.be.revertedWithCustomError(this.token, 'RMRKLocked');
   });
 
+  it('reduces total supply on burn', async function () {
+    await this.token.connect(owner).mint(owner.address, 1);
+    const tokenId = this.token.totalSupply();
+    await this.token.connect(owner)['burn(uint256)'](tokenId);
+    expect(await this.token.totalSupply()).to.equal(0);
+  });
+
+  it('reduces total supply on burn and does not reuse Id', async function () {
+    let tx = await this.token.connect(owner).mint(owner.address, 1);
+    let event = (await tx.wait()).events?.find((e) => e.event === 'Transfer');
+    const tokenId = event?.args?.tokenId;
+    await this.token.connect(owner)['burn(uint256)'](tokenId);
+
+    tx = await this.token.connect(owner).mint(owner.address, 1);
+    event = (await tx.wait()).events?.find((e) => e.event === 'Transfer');
+    const newTokenId = event?.args?.tokenId;
+
+    expect(newTokenId).to.equal(tokenId.add(1));
```

Comment:
I like this approach as well (for checking that the burned token ID is not equal to the newly minted one) 👌